tests/non_matching/mapping_info_02.cc

```diff
@@ -0,0 +1,126 @@
+
+#include <deal.II/grid/filtered_iterator.h>
+#include <deal.II/grid/grid_generator.h>
+
+#include <deal.II/matrix_free/fe_point_evaluation.h>
+
+#include <deal.II/non_matching/fe_values.h>
+#include <deal.II/non_matching/mapping_info.h>
+#include <deal.II/non_matching/mesh_classifier.h>
+#include <deal.II/non_matching/quadrature_generator.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+#include "../tests.h"
+
+void
+test_vec_of_qpoints()
+{
+  constexpr unsigned int dim    = 2;
+  constexpr unsigned int degree = 1;
+
+  FE_Q<dim> fe_q(degree);
+
+  Triangulation<dim> tria;
+
+  MappingQ<dim> mapping(degree);
+
+  GridGenerator::subdivided_hyper_cube(tria, 4);
+
+  DoFHandler<dim> dof_handler(tria);
+
+  dof_handler.distribute_dofs(fe_q);
+
+  Functions::SignedDistance::Sphere<dim> level_set;
+
+  Vector<double> level_set_vec(dof_handler.n_dofs());
+
+  VectorTools::interpolate(dof_handler, level_set, level_set_vec);
+
+  NonMatching::MeshClassifier<dim> mesh_classifier(dof_handler, level_set_vec);
+  mesh_classifier.reclassify();
+
+  hp::QCollection<1> q_collection((QGauss<1>(degree)));
+
+  NonMatching::DiscreteQuadratureGenerator<dim> quadrature_generator(
+    q_collection, dof_handler, level_set_vec);
+
+  // FEPointEvaluation
+  NonMatching::MappingInfo<dim> mapping_info_cell(
+    mapping, update_values | update_gradients | update_JxW_values);
+
+  std::vector<std::vector<Point<dim>>> unit_points_vector;
+  for (const auto &cell : tria.active_cell_iterators())
+    {
+      quadrature_generator.generate(cell);
+      unit_points_vector.emplace_back(
+        quadrature_generator.get_inside_quadrature().get_points());
+    }
+
+  mapping_info_cell.reinit_cells(tria.active_cell_iterators(),
+                                 unit_points_vector);
+
+  Vector<double>      src(dof_handler.n_dofs());
+  std::vector<double> solution_values_in(fe_q.dofs_per_cell);
+
+  for (auto &v : src)
+    v = random_value<double>();
+  for (const auto &cell : dof_handler.active_cell_iterators())
+    {
+      cell->get_dof_values(src,
+                           solution_values_in.begin(),
+                           solution_values_in.end());
+
+      FEPointEvaluation<1, dim, dim, double> fe_point_cell(mapping_info_cell,
+                                                           fe_q);
+      auto                                   test_fe_point_reinit_vec_pts =
+        [&](FEPointEvaluation<1, dim, dim, double> &fe_point) {
+          fe_point.reinit(cell->active_cell_index());
+          fe_point.evaluate(solution_values_in,
+                            EvaluationFlags::values |
+                              EvaluationFlags::gradients);
+          for (const auto q : fe_point.quadrature_point_indices())
+            {
+              deallog << fe_point.get_value(q) << std::endl;
+              deallog << fe_point.get_gradient(q) << std::endl;
+            }
+        };
+
+      test_fe_point_reinit_vec_pts(fe_point_cell);
```

Comment:
The lambda does not really make sense if you only use it for one `FEPointEvaluation` object.

Reply:
You're right! Too much copy-paste, sorry for that